include/treelite/tree_impl.h

```diff
@@ -811,14 +835,14 @@ Model::SerializeTemplate(HeaderPrimitiveFieldHandlerFunc header_primitive_field_
 template <typename HeaderPrimitiveFieldHandlerFunc>
 inline void
 Model::DeserializeTemplate(HeaderPrimitiveFieldHandlerFunc header_primitive_field_handler,
+                           int32_t& major_ver, int32_t& minor_ver, int32_t& patch_ver,
                            TypeInfo& threshold_type, TypeInfo& leaf_output_type) {
-  int major_ver, minor_ver, patch_ver;
   header_primitive_field_handler(&major_ver);
   header_primitive_field_handler(&minor_ver);
   header_primitive_field_handler(&patch_ver);
-  if (major_ver != TREELITE_VER_MAJOR || minor_ver != TREELITE_VER_MINOR) {
+  if (major_ver != TREELITE_VER_MAJOR && !(major_ver == 2 && minor_ver == 4)) {
     std::ostringstream oss;
-    oss << "Cannot deserialize model from a different version of Treelite." << std::endl
+    oss << "Cannot deserialize model from an incompatible version of Treelite." << std::endl
```

Comment:
```suggestion
    oss << "Cannot deserialize model from a different major Treelite version or a version before 2.4.0." << std::endl
```
I know this is _slightly_ incorrect (2.4.0 can be deserialized in 3), but it would be nice to make this error message a little more expressive so that downstream users know more specifically what needs to be done to fix the issue.